internal/topic/topic-operation.go

```diff
@@ -111,6 +115,49 @@ func (operation *Operation) CreateTopics(topics []string, flags CreateTopicFlags
 		topicDetails.ConfigEntries[configParts[0]] = &configParts[1]
 	}
 
+	if flags.File != "" {
+		fileContent, err := os.ReadFile(flags.File)
+		if err != nil {
+			return errors.Wrap(err, "could not read topic description file")
+		}
+
+		fileTopicConfig := Topic{}
+		ext := path.Ext(flags.File)
+		var unmarshalErr error
+		switch ext {
+		case ".yml", ".yaml":
+			unmarshalErr = yaml.Unmarshal(fileContent, &fileTopicConfig)
+		case ".json":
+			unmarshalErr = json.Unmarshal(fileContent, &fileTopicConfig)
+		default:
+			return errors.Wrapf(err, "unsupported file format '%s'", ext)
+		}
+		if unmarshalErr != nil {
+			return errors.Wrap(err, "could not unmarshal config file")
+		}
+
+		numPartitions := int32(len(fileTopicConfig.Partitions))
```

Comment:
@gotha I changed your implementation so that explicitly defined parameters always take precedence over parameters from file

Reply:
ok, cool. 
Thank you for merging this.